syntax/expr_let_test.go

```diff
@@ -49,22 +49,26 @@ func TestExprLetArrayPattern(t *testing.T) {
 
 func TestExprLetTuplePattern(t *testing.T) {
 	AssertCodesEvalToSameValue(t, `4`, `let () = (); 4`)
-	AssertCodesEvalToSameValue(t, `4`, `let (a:x, b:y) = (a:4, b:7); x`)
-	AssertCodesEvalToSameValue(t, `4`, `let (a:x, b:x) = (a:4, b:4); x`)
-	AssertCodesEvalToSameValue(t, `4`, `let x = 4; let (a:x) = (a:4); x`)
-	AssertCodesEvalToSameValue(t, `4`, `let x = 5; let (a:x) = (a:4); x`)
-	AssertCodesEvalToSameValue(t, `4`, `let (a:[x]) = (a:[4]); x`)
+	AssertCodesEvalToSameValue(t, `4`, `let (a: x, b: y) = (a: 4, b: 7); x`)
+	AssertCodesEvalToSameValue(t, `4`, `let (a: x, b: x) = (a: 4, b: 4); x`)
+	AssertCodesEvalToSameValue(t, `4`, `let x = 4; let (a: x) = (a: 4); x`)
+	AssertCodesEvalToSameValue(t, `4`, `let x = 5; let (a: x) = (a: 4); x`)
+	AssertCodesEvalToSameValue(t, `4`, `let (a: [x]) = (a: [4]); x`)
 	AssertCodesEvalToSameValue(t, `1`, `let (:x) = (x: 1); x`)
 	AssertCodesEvalToSameValue(t, `2`, `let (:x, :y) = (x: 1, y: 2); y`)
-	AssertCodeErrors(t, `let (a:x) = (b:7, a:4); x`, "")
-	AssertCodeErrors(t, `let (a:x, a:x) = (a:4, a:4); x`, "")
-	AssertCodeErrors(t, `let (a:x, a:x) = (a:4); x`, "")
-	AssertCodeErrors(t, `let x = 5; let (a:(x)) = (a:4); x`, "")
-	AssertCodeErrors(t, `let (a:x, b:x) = (a:4, b:7); x`, "")
-	AssertCodeErrors(t, `let x = 5; let (a:[(x)]) = (a:[4]); x`, "")
+	AssertCodeErrors(t, `let (a: x) = (b: 7, a: 4); x`, "")
+	AssertCodeErrors(t, `let (a: x, a: x) = (a: 4, a: 4); x`, "")
+	AssertCodeErrors(t, `let (a: x, a: x) = (a: 4); x`, "")
+	AssertCodeErrors(t, `let x = 5; let (a: (x)) = (a: 4); x`, "")
+	AssertCodeErrors(t, `let (a: x, b: x) = (a: 4, b: 7); x`, "")
+	AssertCodeErrors(t, `let x = 5; let (a: [(x)]) = (a: [4]); x`, "")
 }
 
 func TestExprLetDictPattern(t *testing.T) {
+	AssertCodesEvalToSameValue(t, `42`, `let {1: a} = {1: 42}; a`)
+	AssertCodesEvalToSameValue(t, `42`, `let {[1, 2, 3]: a} = {[1, 2, 3]: 42}; a`)
+	AssertCodesEvalToSameValue(t, `42`, `let a = 4; let {"x": a} = {"x": 42}; a`)
+	AssertCodesEvalToSameValue(t, `42`, `let a = 42; let {"x": (a)} = {"x": 42}; a`)
```

Comment:
nit: One or two cases that combine `ident` and `(ident)` would be good. Consider the same for other pattern types.

Reply:
Do you mean `let {"x": a, "y": (a)} = {"x": 42, "y": 42}; a`?

Reply:
Yep, maybe also `b` and `(a)`.

Reply:
Oh, actually, also `let {"x": a, "y": (a)} = {"x": 42, "y": 56}; a` (should equal 42). The `a` is a new binding, while the `(a)` is an existing binding. They should have nothing to do with each other.

Reply:
`let a = 56; let {"x": a, "y": (a)} = {"x": 42, "y": 56}; a` should equal 42?